go/libraries/doltcore/sqle/dolt_index_test.go

```diff
@@ -0,0 +1,1170 @@
+		tests = append(tests, doltIndexTestType{
+			varietyTest.indexName,
+			varietyTest.belowfirstValue,
+			nil,
+		}, doltIndexTestType{
+			varietyTest.indexName,
+			varietyTest.firstValue,
+			[]sql.Row{
+				{int32(-3), uint64(1), forceParseTime(t, "2020-05-14 12:00:00"), "-3.30000", "a", -3.3, "a", "-00:03:03", "a", int16(1980)},
+			},
+		}, doltIndexTestType{
+			varietyTest.indexName,
+			varietyTest.secondValue,
+			[]sql.Row{
+				{int32(-1), uint64(2), forceParseTime(t, "2020-05-14 12:00:01"), "-1.10000", "b", -1.1, "a,b", "-00:01:01", "b", int16(1990)},
+			},
+		}, doltIndexTestType{
+			varietyTest.indexName,
+			varietyTest.thirdValue,
+			[]sql.Row{
+				{int32(0), uint64(3), forceParseTime(t, "2020-05-14 12:00:02"), "0.00000", "c", 0.0, "c", "00:00:00", "c", int16(2000)},
+			},
+		}, doltIndexTestType{
+			varietyTest.indexName,
+			varietyTest.fourthValue,
+			[]sql.Row{
+				{int32(1), uint64(4), forceParseTime(t, "2020-05-14 12:00:03"), "1.10000", "d", 1.1, "a,c", "00:01:01", "d", int16(2010)},
+			},
+		}, doltIndexTestType{
+			varietyTest.indexName,
+			varietyTest.fifthValue,
+			[]sql.Row{
+				{int32(3), uint64(5), forceParseTime(t, "2020-05-14 12:00:04"), "3.30000", "e", 3.3, "b,c", "00:03:03", "e", int16(2020)},
+			},
+		}, doltIndexTestType{
+			varietyTest.indexName,
+			varietyTest.abovefifthValue,
+			nil,
+		})
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("%s|%v", test.indexName, test.keys), func(t *testing.T) {
+			index, ok := indexMap[test.indexName]
+			require.True(t, ok)
+			testDoltIndex(t, test.keys, test.expectedRows, index.Get)
+		})
+	}
+}
+
+func TestDoltIndexGreaterThan(t *testing.T) {
+	indexMap := doltIndexSetup(t)
+
+	tests := []struct {
+		indexName    string
+		keys         []interface{}
+		expectedRows []sql.Row
+	}{
+		{
+			"onepk:primaryKey",
+			[]interface{}{1},
+			[]sql.Row{{2, 1, 2}, {3, 3, 3}, {4, 4, 3}},
+		},
+		{
+			"onepk:primaryKey",
+			[]interface{}{3},
+			[]sql.Row{{4, 4, 3}},
+		},
+		{
+			"onepk:primaryKey",
+			[]interface{}{4},
+			nil,
+		},
+		{
+			"onepk:idx_v1",
+			[]interface{}{1},
+			[]sql.Row{{3, 3, 3}, {4, 4, 3}},
+		},
+		{
+			"onepk:idx_v1",
+			[]interface{}{3},
+			[]sql.Row{{4, 4, 3}},
+		},
+		{
+			"onepk:idx_v1",
+			[]interface{}{4},
+			nil,
+		},
+		{
+			"twopk:primaryKey",
+			[]interface{}{1, 1},
+			[]sql.Row{{1, 2, 3, 4}, {2, 1, 4, 4}, {2, 2, 4, 3}},
+		},
+		{
+			"twopk:primaryKey",
+			[]interface{}{2, 1},
+			[]sql.Row{{2, 2, 4, 3}},
+		},
+		{
+			"twopk:idx_v2v1",
+			[]interface{}{3, 4},
+			[]sql.Row{{1, 2, 3, 4}, {2, 1, 4, 4}},
+		},
+		{
+			"twopk:idx_v2v1",
+			[]interface{}{4, 3},
+			[]sql.Row{{2, 1, 4, 4}},
+		},
+	}
+
+	for _, varietyTest := range varietyTests {
+		tests = append(tests, doltIndexTestType{
+			varietyTest.indexName,
+			varietyTest.belowfirstValue,
+			[]sql.Row{
+				{int32(-3), uint64(1), forceParseTime(t, "2020-05-14 12:00:00"), "-3.30000", "a", -3.3, "a", "-00:03:03", "a", int16(1980)},
```

Comment:
Suggest extracting vars for all these variety table rows (typesTableRow1, typesTableRow2, etc) for all these tests. Would be easier to read.